src/nwbinspector/checks/nwb_containers.py

```diff
@@ -28,6 +28,25 @@ def check_large_dataset_compression(nwb_container: NWBContainer, gb_lower_bound:
             )
 
 
+@register_check(importance=Importance.BEST_PRACTICE_VIOLATION, neurodata_type=NWBContainer)
+def check_dataset_is_empty(nwb_container: NWBContainer):
+
+    def is_empty_data(data):
+        if isinstance(data, ndarray):
+            return data.size == 0
+        elif isinstance(data, (list, tuple)):
+            return all(is_empty_data(item) for item in data)
+        else:
+            return not bool(data)
```

Comment:
I wouldn't mind if this were moved to the common `utils` to be used by any other check (or anyone else) going forward 😃